### PR TITLE
Feature/np 2382 expose cluster stats

### PR DESCRIPTION
**Pull request by Thiamath**
#### What does this PR do?

- Create the `monitoring` domain on the `public-api-cli`.
- Wire up the `cluster-stats` and `cluster-summary` from the monitoring-manager to `monitoring` command domain.

#### What is missing?
This PR is blocked by [this other one](https://github.com/nalej/monitoring/pull/32).

#### What are the relevant tickets?

- [NP-2382](https://nalej.atlassian.net/browse/NP-2382)
- [NP-2368](https://nalej.atlassian.net/browse/NP-2368)

#### Screenshots
Default__bash_
Default__bash_

### CODE REVIEW

File: cmd/public-api-cli/commands/monitoring.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	monitoringCmd.AddCommand(clusterSummaryCmd)
+	clusterSummaryCmd.Flags().Int32Var(&rangeMinutes, "range-minutes", 0, "Return average values over the past <rangeMinutes> minutes.")
+}
+
+var clusterStatsCmd = &cobra.Command{
+	Use:   "cluster-stats [clusterID]",
+	Short: "Display the cluster stats",
+	Long:  `Display the cluster stats`,
+
+	Args: cobra.MinimumNArgs(1),
```
Comment (dhiguero) [inline cmd/public-api-cli/commands/monitoring.go]:
```suggestion
	Args: cobra.ExactArgs(1),
```

File: cmd/public-api-cli/commands/monitoring.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		monitoring.GetClusterStats(cliOptions.Resolve("organizationID", organizationID), args[0], rangeMinutes, clusterStatFields)
+	},
+}
+
+var clusterSummaryCmd = &cobra.Command{
+	Use:   "cluster-summary [clusterID]",
+	Short: "Display the cluster stats summary",
+	Long:  `Display the cluster stats summary`,
+
+	Args: cobra.MinimumNArgs(1),
```
Comment (dhiguero) [inline cmd/public-api-cli/commands/monitoring.go]:
```suggestion
	Args: cobra.ExactArgs(1),
```

File: cmd/public-api-cli/commands/monitoring.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		SetupLogging()
+		_ = cmd.Help()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(monitoringCmd)
+
+	monitoringCmd.AddCommand(clusterStatsCmd)
+	clusterStatsCmd.Flags().Int32Var(&rangeMinutes, "range-minutes", 0, "Return average values over the past <rangeMinutes> minutes.")
```
Comment (dhiguero) [inline cmd/public-api-cli/commands/monitoring.go]: I think the parameter should be named `rangeMinutes` to match other command flags naming convention. Also, given that it is a flag, it should have a default value, or document how 0 is interpreted.

File: internal/app/cli/edge-monitoring.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	End        string
+	Resolution time.Duration
+}
+
+func dateParse(in string) int64 {
+	if in == "" {
+		return 0
+	}
+
+	t, err := dateparse.ParseLocal(in)
```
Comment (dhiguero) [inline internal/app/cli/edge-monitoring.go]: Why are we using an extra dependency and not the golang `time` package?

File: internal/app/cli/monitoring.go [hunk 5]
```diff
@@ -12,139 +12,115 @@
...
+				case 1:
+					platformStatFields = append(platformStatFields, grpc_monitoring_go.PlatformStatsField_VOLUMES)
+				case 2:
+					platformStatFields = append(platformStatFields, grpc_monitoring_go.PlatformStatsField_FRAGMENTS)
+				case 3:
+					platformStatFields = append(platformStatFields, grpc_monitoring_go.PlatformStatsField_ENDPOINTS)
+				}
+			} else {
+				log.Warn().Str("field", fieldName).Msg("Field name does not exist and will be ignored.")
+			}
```
Comment (dhiguero) [inline internal/app/cli/monitoring.go]: Use the map value with casting, as this switch may break inadvertently if the protos change.

File: cmd/public-api-cli/commands/monitoring.go (hunk 3 above)
Comment (Thiamath) [reply cmd/public-api-cli/commands/monitoring.go]: I'm ok with that. Somehow I don't like camel case in console params (even when I really love camel case) but I agree that it is the Nalej convention.

File: internal/app/cli/edge-monitoring.go (hunk 4 above)
Comment (Thiamath) [reply internal/app/cli/edge-monitoring.go]: I just renamed the file, not sure about what's inside.
But I can spend some time refactoring it if you think it worths.

Comment (aagea) [reply internal/app/cli/edge-monitoring.go]: We must understand the usage of this dependency and remove it if it is not necessary.

Comment (Thiamath) [reply internal/app/cli/edge-monitoring.go]: As I've seen [here](https://github.com/nalej/public-api/blob/c8e38e14da820c742a1e9b55ec8830886dd52a5b/cmd/public-api-cli/commands/inventory.go#L52), the CLI is getting some date/time params from the console.
The `dateparse` library is being used to parse those params no matter what format was used to specify them.
We could specify a constraint (TBD) for the format used on those params to enforce a single pattern and avoid the need to use that dependency.

Comment (dhiguero) [reply internal/app/cli/edge-monitoring.go]: Ok, seems more user-friendly for the CLI

**Pull request merged**